app/controllers/welcome_controller.rb

```diff
@@ -4,95 +4,17 @@ def index
-                               stats_saving_throws_throws_cha: base_info.stats.saving_throws.throws[:cha],
-                               stats_skills_proficiencies: base_info.stats.skills.proficiencies.to_s,
-                               stats_skills_acrobatics: base_info.stats.skills.skills[:acrobatics],
-                               stats_skills_animal_handling: base_info.stats.skills.skills[:animal_handling],
-                               stats_skills_arcana: base_info.stats.skills.skills[:arcana],
-                               stats_skills_athletics: base_info.stats.skills.skills[:athletics],
-                               stats_skills_deception: base_info.stats.skills.skills[:deception],
-                               stats_skills_history: base_info.stats.skills.skills[:history],
-                               stats_skills_insight: base_info.stats.skills.skills[:insight],
-                               stats_skills_intimidation: base_info.stats.skills.skills[:intimidation],
-                               stats_skills_investigation: base_info.stats.skills.skills[:investigation],
-                               stats_skills_medicine: base_info.stats.skills.skills[:medicine],
-                               stats_skills_nature: base_info.stats.skills.skills[:nature],
-                               stats_skills_perception: base_info.stats.skills.skills[:perception],
-                               stats_skills_performance: base_info.stats.skills.skills[:performance],
-                               stats_skills_persuasion: base_info.stats.skills.skills[:persuasion],
-                               stats_skills_religion: base_info.stats.skills.skills[:religion],
-                               stats_skills_sleight_of_hand: base_info.stats.skills.skills[:sleight_of_hand],
-                               stats_skills_stealth: base_info.stats.skills.skills[:stealth],
-                               stats_skills_survival: base_info.stats.skills.skills[:survival],
-                               sub_ancestry: base_info.sub_ancestry.instance_of?(String) ? base_info.sub_ancestry : base_info.sub_ancestry[:name],
-                               traits: base_info.traits.to_s,
-                               vision: base_info.vision,
-                               is_saved: 0)
-        NpcModelPack.create(npc_model_id: @npc.id, pack_id: base_info.equipment.pack.id)
-        base_info.equipment.armor.each do |piece|
-          NpcModelArmor.create(npc_model_id: @npc.id, armor_id: piece.id) unless piece == ''
-        end
-        base_info.equipment.weapons.each do |weapon|
-          NpcModelWeapon.create(npc_model_id: @npc.id, weapon_id: weapon.id)
-        end
-        @armor = base_info.equipment.armor
-        @pack = [base_info.equipment.pack]
-        @weapons = base_info.equipment.weapons
-      end
+      base_info = NPCFacade.create_npc(params[:post][:score_type].downcase, params[:post][:level].to_i)
+      @npc = base_info[0]
```

Comment:
I wanted the visitor and user to use the same page to display the same thing, which means even visitors will create NpcModels.  They don't have a way to save the NpcModel, so if they refresh, the old one is destroyed.  If they leave, whoever visits the page next will destroy the last NPC.  If no one ever visits again, I guess we've got bigger problems than one extra, floating NPC in the DB.